sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -989,7 +989,7 @@ class Analyzer(
               withPosition(u) {
                 try {
                   outer.resolve(nameParts, resolver) match {
-                    case Some(outerAttr) => OuterReference(outerAttr)
+                    case Some(outerAttr) => OuterReference(outerAttr)()
```

Comment:
Another interesting case to consider:
````sql
select ...
from   t1
where  t1.c1 in (select sum(t1.c2) from t2)
````

If we support correlated columns in SELECT clause, do we build the Aggregate on T2 or T1?